### PR TITLE
Set fields on orphan cleanup task

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #4662
https://pulp.plan.io/issues/4662

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/en/3.0/nightly/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/models/task.py [hunk 1]
```diff
@@ -160,6 +160,17 @@ def with_reservations(self, resources):
...
+    def resource_managers(self):
+        """
+        Returns a queryset of resource manager workers. Resource managers are identified based on
+        their name. Note that some of these may be offline.
+
+        Returns:
+            :class:`django.db.models.query.QuerySet`:  A query set of the Worker objects which
+                which match the resource manager name.
+        """
+        return self.filter(name__startswith=TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME)
```
Comment (daviddavis) [inline pulpcore/app/models/task.py]: I ended up not needing this change but it seems useful..?

File: pulpcore/app/models/task.py [hunk 2]
```diff
@@ -160,6 +160,17 @@ def with_reservations(self, resources):
         """
         return self.filter(reservations__resource__in=resources).distinct().get()
 
+    def resource_managers(self):
+        """
+        Returns a queryset of resource manager workers. Resource managers are identified based on
```
Comment (bmbouter) [inline pulpcore/app/models/task.py]: nitpick I believe the docstring PEP specifies multiline docstrings to have the paragraph be a single line.

File: pulpcore/tasking/tasks.py [hunk 3]
```diff
@@ -83,13 +83,14 @@ def _queue_reserved_task(func, inner_task_id, resources, inner_args, inner_kwarg
                 time.sleep(0.25)
                 continue
             else:
-                task_status.state = TASK_STATES.RUNNING
-                task_status.save()
+                rq_worker = util.get_current_worker()
+                worker = Worker.objects.get(name=rq_worker.name)
+                task_status.worker = worker
+                task_status.set_running()
```
Comment (bmbouter) [inline pulpcore/tasking/tasks.py]: :+1:

File: pulpcore/tasking/util.py [hunk 4]
```diff
@@ -92,3 +93,17 @@ def get_url(model):
...
     """
     return reverse(get_view_name_for_model(model, 'detail'), args=[model.pk])
+
+
+def get_current_worker():
+    """
+    Get the rq worker assigned to the current job
+
+    Returns:
+       rq.worker.Worker: The worker assigned to the current job
```
Comment (bmbouter) [inline pulpcore/tasking/util.py]: Optional idea:  Maybe have this use to make the object stylized in Sphinx prettier.

```
class:`rq.worker.Worker`
```

**Comment by bmbouter** [comment]
Can a CHANGES bugfix entry be added?

**Pull request merged**